includes/Widgets/TopratedProducts.php

```diff
@@ -28,7 +28,9 @@ public function __construct() {
      * @param array $instance Saved values from database.
      */
     public function widget( $args, $instance ) {
-        $title = apply_filters( 'widget_title', $instance['title'] );
+        $title         = isset( $instance['title'] ) ? apply_filters( 'widget_title', $instance['title'] ) : '';
+        $no_of_product = isset( $instance['no_of_product'] ) ? $instance['no_of_product'] : 8;
+        $show_rating   = isset( $instance['show_rating'] ) ? $instance['show_rating'] : false;
         extract( $instance );
```

Comment:
do we really need this extract here? this will overwrite above title, no_of_product and show_rating variable.